frontend/src/js/controllers/changePwdCtrl.js

```diff
@@ -7,67 +7,87 @@
             parameters.method = 'POST';
             parameters.data = {
                 "old_password": vm.user.old_password,
                 "new_password1": vm.user.new_password1,
                 "new_password2": vm.user.new_password2,
+                "uid": $state.params.user_id,
             }
-            parameters.token = userKey;
-            var passwordCheck = validationSvc.password_check(parameters.data.new_password1, parameters.data.new_password2);
-            if (passwordCheck.status) {
                 parameters.callback = {
                     onSuccess: function(response) {
                         var status = response.status;
                         var response = response.data;
                         vm.user.error = false;
                         console.log("PASSWORD CHANGED SUCCESSFULLY");
                         console.log(response);
+                        vm.stopLoader();
                         // navigate to challenge page
                         // $state.go('web.challenge-page.overview');
                     },
                     onError: function(response) {
                         var status = response.status;
                         var error = response.data;
                         vm.user.error = "Failed";
-                        if (status == 400) {
-                            console.log("ERROR Occured");
-                            console.log(error);
-                            angular.forEach(error, function(value, key) {
-                                if (key == 'old_password') {
-                                    vm.isValid.old_password = true;
-                                    vm.wrnMsg.old_password = value[0];
-                                }
-                                if (key == 'new_password1') {
-                                    vm.isValid.new_password1 = true;
-                                    vm.wrnMsg.new_password1 = value[0];
-                                }
-                                if (key == 'new_password2' || key == 'non_field_errors') {
-                                    vm.isValid.new_password2 = true;
-                                    vm.wrnMsg.new_password2 = value[0];
-                                }
-                            })
+                        vm.isFormError = true;
+                        var oldpassword_valid ,password1_valid, password2_valid;
+                        try {
+                            oldpassword_valid = typeof(response.data.old_password) !== 'undefined' ? true : false;
+                            password1_valid = typeof(response.data.new_password1) !== 'undefined' ? true : false;
+                            password2_valid = typeof(response.data.new_password2) !== 'undefined' ? true : false;
+                            if (oldpassword_valid) {
+                                vm.FormError = response.data.oldpassword_valid[0] + " " + response.data.oldpassword_valid[1];
```

Comment:
this  
`vm.FormError = response.data.oldpassword_valid[0] + " " + response.data.oldpassword_valid[1];`
must be
`vm.FormError = response.data.old_password[0] + " " + response.data.old_password[1];`